### ISSUE TITLE
RaspberryPI

### ISSUE THREAD

**Issue by ankh2054** [OstlerDev/PopcornTV#239]
Hi,

When trying to load a movie the following error occurs.

error: Error: Can't load the addon. Issue to: https://github.com/PopcornTV/node-images/issues Error: Cannot find module './bindings/linux/arm/0.12.0/binding.node'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/home/pi/PopcornTV/node_modules/ptv-images/binding.js:52:34)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/home/pi/PopcornTV/node_modules/ptv-images/binding.js:55:23)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/home/pi/PopcornTV/node_modules/ptv-images/index.js:4:15)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)

**Comment by ankh2054** [comment]
Will try this solution tonight - https://github.com/OstlerDev/PopcornTV/issues/223

**Comment by ankh2054** [comment]
Error has changed to the following:

<code>Error: Can\'t load the addon. Issue to: https://github.com/PopcornTV/node-images/issues Error: /home/pi/PopcornTV/node_modules/ptv-images/bindings/linux/arm/0.12.0/binding.node: invalid ELF header'M</code>

**Comment by OstlerDev** [comment]
Did you follow the guide on the wiki? https://github.com/OstlerDev/PopcornTV/wiki/Running-PopcornTV-on-a-VPS

**Comment by ankh2054** [comment]
Yes and have the following versions installed.
node = 0.12.6
npm = 2.11.2

**Comment by gpsmit** [comment]
I have the same issue, and followed the link on the wiki.
I tried node 4.0.0, node 0.12.6 and node 0.10.4
All with the same issue

I came across this issue as well on the osx, there it was enough to revert from 4.0 to 0.12, sadly on Rpi this doesnt solve the issue. I hope you can find out what it is this (with @Pizara 's help?)
PopcornTV is awesome, thanks for everything

**Comment by gpsmit** [comment]
I managed to get it working on my Raspberry Pi 2
What i did: I installed node **0.12.0** by compiling it on my Pi 2:
```
wget http://nodejs.org/dist/v0.12.0/node-v0.12.0.tar.gz
tar xvf node-v0.12.0.tar.gz
cd node-v0.12.0
./configure
make
sudo make install
```

For older raspberrys compiling it on your pi is to intensive and i think there are precompiled 0.12.0 versions online... I do not know if these will work but its work though:
http://conoroneill.net/download-compiled-version-of-nodejs-0120-stable-for-raspberry-pi-here

I then updated npm to the latest npm version (dont know if this is needed, but doesnt hurt i guess).
Next i followed the usual Mac install steps for PopcornTV followed by @Pizara's steps linked above in this thread.
I did also change the ownership of the new binding.node file to pi, just to be sure.

In short. Pizara's binding.node file in combination with node 0.12.0 works.
Thanks so much dude!
 
I have not tested this binding.node file with other node versions

**Comment by masonjames** [comment]
Hey @ankh2054 did you get it working?

The binding.node and node-images errors were the most difficult part for my build as well. I've had it running on a b+ model for over a month without many problems now.  Kinda scared to update stuff, but happy it works :)

**Comment by ankh2054** [comment]
Hi Mason,

Not yet, I'm going to recheck permissions today. I know the feeling of not wanting to update haha. I feel Like I'm so close now. 

Charles Holtzkampf (Baz)

**Comment by ankh2054** [comment]
permissions are all fine, so going to try and fid out what this invalid ELF header means. OR perhaps I should start over, as I did so many things to get to this point I might of caused the issue along the way.

**Comment by gpsmit** [comment]
Yea thats what i did... just started from scratch again, deleted node and ptv. the compiling of 0.12.0 took the longest. afterwards its business as usual

**Comment by ankh2054** [comment]
very odd, it works for the first movie after which I stat getting the same error until I reboot. I think I might start from scratch by re-installing Raspbian.

**Comment by masonjames** [comment]
I'm still using Raspbian Wheezy with node.js v0.12.6 and everything works well.

I just got a model 2 so I may attempt another fresh install with Jesse & nodejs 4.1.2.  I'm also curious to see if performance is greatly improved on the new model.

An ELF error usually occurs when the binary file isn't compiled for the sever's platform.  In the guide I reference using this version of nodejs because it was compiled specifically for the (now older) raspberry pi:
http://conoroneill.net/node-v0122-for-arm-v6v7-including-raspberry-pi-raspberry-pi-2-and-odroid-c1/
